src/Tizen.MachineLearning.Service/Tizen.MachineLearning.Service/Service.cs

```diff
@@ -0,0 +1,696 @@
+        {
+            if (handle == IntPtr.Zero)
+                throw MLServiceExceptionFactory.CreateException(MLServiceError.InvalidParameter, "The service handle is null");
+
+            _handle = handle;
+
+            _eventCallback = (type, event_data_handle, _) =>
+            {
+                if (type == EventType.NewData && EventReceived!= null) {
+                    MlInformation info = new MlInformation(event_data_handle);
+                    EventReceived?.Invoke(this, new ServiceReceivedEventArgs(info));
+                }
+            };
+
+            MLServiceError ret = Interop.Service.SetEventCb(_handle, _eventCallback, IntPtr.Zero);
+            MLService.CheckException(ret, "Failed to register event callback");
+        }
+
+        /// <summary>
+        /// Destroys the MlInformation resource.
+        /// </summary>
+        /// <since_tizen> 12 </since_tizen>
+        ~Service()
+        {
+            Dispose(false);
+        }
+
+        internal IntPtr GetHandle()
+        {
+            return _handle;
+        }
+
+        /// <summary>
+        /// Starts the process of service instance.
+        /// </summary>
+        /// <feature>http://tizen.org/feature/machine_learning.service</feature>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed to start the service.</exception>
+        /// <since_tizen> 12 </since_tizen>
+        public void Start()
+        {
+            MLServiceError ret = Interop.Service.Start(_handle);
+            MLService.CheckException(ret, "Failed to start the service because of internal error");
+        }
+
+        /// <summary>
+        /// Stops the process of service instance.
+        /// </summary>
+        /// <feature>http://tizen.org/feature/machine_learning.service</feature>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed to stop the service.</exception>
+        /// <since_tizen> 12 </since_tizen>
+        public void Stop()
+        {
+            MLServiceError ret = Interop.Service.Stop(_handle);
+            MLService.CheckException(ret, "Failed to stop the service because of internal error");
+        }
+
+        /// <summary>
+        /// Gets the TensorsInfo of the service instance.
+        /// </summary>
+        /// <param name="IsInput">The boolean value for choosing input or output to get.</param>
+        /// <param name="name">The name of node in the pipeline. You can set NULL if serivce is constructed from model configuration.</param>
```

Comment:
The name of node in the pipeline. You can set empty string if machine learning service is constructed from model configuration.

Reply:
Updated!